pkg/lib/dialogs.jsx

```diff
@@ -115,10 +117,26 @@ export const WithDialogs = ({ children }) => {
                               JSON.stringify(dialog));
             is_open.current = !!component;
             setDialog(component);
+            return new Promise((resolve, reject) => {
+                setResolvePromise(() => resolve);
```

Comment:
This triggers a re-render, no? Could we store the promises in `useRef` things? Maybe we can use a single useRef to store all of `is_open`, `resolvePromise`, and `rejectPromise`.
Actually, resolvePromise != null can take the place of is_open, no?

Reply:
It calls a `setState` so it re-triggers state yes. But they should be queued with other setState's so this should be the same and we already trigger a re-render for `setDialog`

Reply:
Good point.